### DISCUSSION TITLE
Grouping :global styles within a scoped selector

### DISCUSSION THREAD

**Discussion by mayank99** [withastro/roadmap#227] [Proposal]
Astro's scoped styles are usually great! But sometimes we need to nest some global styles under a scoped selector. This is often the case when trying to style an external component that is not part of the current one. Today we can do that like so:
```css
.parent { ... }
.parent :global(.child) { ... }
.parent :global(.child .grandchild) { ... }
```

This gets repetitive/tedious really quickly. In CSS modules this is easy to solve using a preprocessor like Sass. Their [docs](https://github.com/css-modules/css-modules#usage-with-preprocessors) even recommend using preprocessors for grouping global styles under `:global`.
```scss
.parent {
  // ...

  :global {
    .child {
      // ...

      .grandchild {
        // ...
      }
  }
}
```

Today this does not work in Astro. When `:global` is used without a selector inside the parentheses, it fails silently and gets treated as `global` (without colons). So the above snippet results in the following:
```css
.parent { ... }
.parent global .child { ... }
.parent global .child .grandchild { ... }
```

If this feature was supported, it would also open up other interesting opportunities, such as Sass mixins:
```html
<style lang='scss'>
  .parent {
    :global {
      @include separator-between-elements;
    }
  }

  @mixin separator-between-elements {
    > * + * {
      border-top: 1px solid;
    }
  }
</style>
```

---

<sup>_Previously I opened https://github.com/withastro/astro/issues/3929 in the main astro repo because I thought it was a bug but apparently it's a missing feature, hence this RFC._</sup>

**Comment by mayank99** [comment]
This is not "resolved", is it?

**Reply by florian-lefebvre** [reply]
Sorry I was purging the roadmap so I should have closed this as outdated. But since there's still interest, I'll reopen!

**Reply by mayank99** [reply]
Thinking more on this, while it would be nice if this was supported, it's not a big deal. There are other more important styling-related problems I would like to see solved in Astro. So I'll close it myself.
